### PR TITLE
Add Go 1.17 to the tests and release

### PR DESCRIPTION
**Pull request by bentranter**
Adds Go 1.17 to our test matrix, and uses it to build the release.

### CODE REVIEW

**Comment by bentranter** [comment]
Oh no lol I didn't mean to update the `/vendor` directory with the `gofmt` stuff 😅 I'll fix it

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [ 1.15.x, 1.16.x]
+        go-version: [ 1.15.x, 1.16.x, 1.17.x ]
```
Comment (andrewsomething) [inline .github/workflows/ci.yml:13]: How do we feel about dropping 1.15? Feel like we should try to align to the upstream policy when possible: 

https://golang.org/doc/devel/release#policy

Double checked and the Snap is already on 1.16:

https://github.com/digitalocean/doctl/blob/283ec460997e86b99a09a2bd4ba223636eece782/dockerfiles/Dockerfile.snap#L49

**Review by andrewsomething — APPROVED** [review]
:+1: Looks good!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andrewsomething) [reply .github/workflows/ci.yml:13]: On the other hand, it doesn't hurt to know if it passes or not... Maybe just mark as not required?

**Review by ChiefMateStarbuck — APPROVED** [review]

**Pull request merged**